ci-operator/step-registry/gcp/provision/workload-identity/gcp-provision-workload-identity-commands.sh

```diff
@@ -0,0 +1,82 @@
+
+cred_req_d=`mktemp -d`
+sts_res_d=`mktemp -d`
+
+echo -e "\n$(date -u --rfc-3339=seconds) - (1/4) Extracting credentials-requests from image ${RELEASE_IMAGE_LATEST}"
+CMD="oc adm release extract ${RELEASE_IMAGE_LATEST} --credentials-requests --cloud=gcp --to ${cred_req_d}"
+run_command "${CMD}" || exit 1
+echo -e "\n$(date -u --rfc-3339=seconds) - The extracted GCP credentials-requests: "
+ls "${cred_req_d}" || exit 1
+
+pushd "/tmp" # the 'ccoctl' would be put in /tmp dir, as no permission in the default dir
+echo -e "\n$(date -u --rfc-3339=seconds) - (2/4) Extracting the CCO utility (ccoctl) binary"
+cco_image=$(oc adm release info --image-for="cloud-credential-operator" "${RELEASE_IMAGE_LATEST}")
+CMD="oc image extract -a ${PULL_SECRET_PATH} ${cco_image} --file='/usr/bin/ccoctl'"
+run_command "${CMD}" || exit 1
+chmod +x ./ccoctl
+
+echo -e "\n$(date -u --rfc-3339=seconds) - (3/4) Creating workload identity and IAM roles"
+CMD="./ccoctl gcp create-all --name=${prefix} --project ${PROJECT_NAME} --region=${REGION} --credentials-requests-dir=${cred_req_d} --output-dir ${sts_res_d}"
+run_command "${CMD}" || exit 1
+
+# https://docs.openshift.com/container-platform/4.10/authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.html#cco-ccoctl-gcp-image-registry_cco-mode-gcp-workload-identity
+echo -e "\n$(date -u --rfc-3339=seconds) - (4/4) Creating long-live key for image-registry (https://docs.openshift.com/container-platform/4.10/authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.html#cco-ccoctl-gcp-image-registry_cco-mode-gcp-workload-identity)"
+image_registry_credential_yaml="${sts_res_d}/manifests/openshift-image-registry-installer-cloud-credentials-credentials.yaml"
+image_registry_sa=$(gcloud iam service-accounts list --filter="displayName=${prefix}-openshift-image-registry-gcs" --format=json | /tmp/jq -r '.[].email')
+
+new_key_json="image_registry_key.json"
+CMD="gcloud iam service-accounts keys create ${new_key_json} --iam-account=${image_registry_sa}"
+run_command "${CMD}" || exit 1
+new_key_str_b64=$(cat "${new_key_json}" | base64 -w 0)
+
+/tmp/yq r -j "${image_registry_credential_yaml}" > tmp1.json
+/tmp/jq --arg k "${new_key_str_b64}" '.data["service_account.json"] = $k' < tmp1.json > tmp2.json
+cat tmp2.json | /tmp/yq r -P - > "${image_registry_credential_yaml}"
+popd
+
+while IFS= read -r -d '' item
+do
+  manifest="$( basename "${item}" )"
+  cp "${item}" "${SHARED_DIR}/manifest_${manifest}"
+done <   <( find "${sts_res_d}"/manifests \( -name "*.yml" -o -name "*.yaml" \) -print0)
+
+while IFS= read -r -d '' item
+do
+  manifest="$( basename "${item}" )"
+  cp "${item}" "${SHARED_DIR}/tls_${manifest}"
+done <   <( find "${sts_res_d}"/tls \( -name "*.key" -o -name "*.pub" \) -print0)
+
+echo -e "\n$(date -u --rfc-3339=seconds) - Manifests & tls keys/pubs were copied to SHARED_DIR"
+ls "${SHARED_DIR}"/manifest_* "${SHARED_DIR}"/tls_*
```

Comment:
I see dev have more simple commands for manifests copying part [ipi-conf-gcp-oidc-creds-provision-commands.sh](https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/conf/gcp/oidc-creds-provision/ipi-conf-gcp-oidc-creds-provision-commands.sh#L22-L27), what do you think which one is better?